### DISCUSSION TITLE
GitHub's advanced search sucks

### DISCUSSION THREAD

**Discussion by JohnReedLOL** [community/community#45250] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I'm trying to use this: [https://github.com/search/advanced](https://github.com/search/advanced)

My goal is to contribute to open source but I have some requirements on my search. I have some cognitive issues so I'd like the codebase that I contribute to to be relatively small so I can read and understand it all. Also, I'd like the codebase to be written in Rust because I am learning that programming language and I like it. Also, I'd like the open source issues I work on to be beginner friendly and easy. Additionally, I'd like the repository I'm contributing to to be relatively popular so that my contribution has real world impact. Finally, I'd like the codebase that I contribute to be actively maintained recently and for the issues that I address on it to be relatively recent.

To try and find GitHub issues that address all these requirements, I put the following into the search:

- size:<1000
- language:Rust
- label:easy
- stars:>1000
- pushed:>2021-01-01
- updated:>2021-01-01

I click on "Issues" and I have [this search](https://github.com/search?q=stars%3A%3E1000+size%3A%3C1000+pushed%3A%3E2021-01-01+label%3Aeasy+updated%3A%3E2021-01-01+language%3ARust&type=issues) which says there are 516 issues that match my search.

Now, there are some things wrong with this search. The first issue is [this](https://github.com/dginev/ar5iv/issues/309). It's an issue in a repository that has 400 stars. I intended for the repository the issue is in to have a 1000 star minimum, so that's not good. And that's what I got when the results were sorted by "Best match". Now I try making a change. Maybe I'll consider allowing issues from repositories of a size that is ">1000" instead of "<1000". I make that change in my search, which now looks like [this](https://github.com/search?q=stars%3A%3E1000+size%3A%3E1000+pushed%3A%3E2021-01-01+label%3Aeasy+updated%3A%3E2021-01-01+language%3ARust&type=issues).

There's a problem. It still says 516 results, and the first result is still [this](https://github.com/dginev/ar5iv/issues/309). The issue search totally disregarded my specifications for how many stars I wanted the repository the issue is in to have and the size of that repository to be. I guess that makes sense because those are under "Repositories options" and not "Issues options". That being said, I want the requirements I set on the repository to apply to the repository that the issue I'm searching for is in. Like if my search has both "stars:>1000" and "label:easy", that means I want issues with the "easy" label that are inside a repository that has over 1000 stars. The advanced search isn't intuitive. Like it isn't obvious by looking at [the advanced search](https://github.com/search/advanced) that the options under "Advanced options" can apply to everything (repository, issues, users, etc) but the options under "Repositories options" only apply when we're looking at "repository results" and are ignored otherwise. I also don't like how the number of results is often "fuzzy" or "approximate", changing up and down from search to search. For example, if I just search "language:Rust" and refresh it a few times, I get "Showing 49,574 available repository results", then "Showing 51,907 available repository results", then "Showing 48,320 available repository results". That doesn't make any sense - there is no way the number of Rust repositories increased by 2 thousand than decreased by over 3 thousands within the span of a second.

Can you please make the advanced search better? Personally, when I go to [https://github.com/search/advanced](https://github.com/search/advanced), I'd like to have separate links in there for "https://github.com/search/advanced/repositories" and "https://github.com/search/advanced/issues". In the issues advanced search, I'd like to be able to specify options regarding the repository of the issue that I'm searching for. Like I'd like to be able to say "I want the issue to be in a repository that has over 1000 stars".

**Comment by divStar** [comment]
Oh my god.. I use GitHub AND my company uses GitHub Enterprise and GitHub still can't manage to implement a search, that actually deserves to be called a search?

Please, GitHub, if you're not smart enough to do that, hire some qualified developers (such as myself). We / they will definitely be able to implement a proper search, that's not a complete embarrassment.

How is it possible, that after 6+ years GitHub still doesn't have a proper search. In what world would a code management site NOT allow to include special characters (such as slashes, backslashes etc.) in their search? Just do a URI-encode / -decode, have the users use single or double quotes or something to escape special characters if need be. But simply ignoring them? On a site focused around code, which typically includes various special characters? That's outright dumb.

Higher proper product owners and developers to repair a broken product you charge a lot of money for!

**Reply by JohnReedLOL** [reply]
Hey. This issue with the advanced search is getting in the way of me being able to contribute to open source. Is there anyone you can talk to or anywhere else we can post to get this fixed?
